### PR TITLE
fix(cli): fixing bug related to flag -q + adding new cli flag related to library path

### LINKED ISSUE #3895: Providing local queries requires library.rego present
### Expected Behavior
When passing `--queries-path` with local queries, the queries loaded
### Actual Behavior
Queries failed to load. Only on log-level=DEBUG the error is shown and requires `library.rego` to present in a specific location.

The location itself is also not documented and not easy to understand where should we place the file. 
I would like to have this file optional.
### Steps to Reproduce the Problem

1.  Write query in some local folder
2.  Run `kics scan --queries-path /path/to/local/folder`
3.  Scan results shows no queries were loaded

### Specifications

-   Version: v1.3.3
-   Platform: Docker
-   Subsystem: Alpine

### PR DESCRIPTION
**Pull request by rafaela-soares**
Closes #3895

**Proposed Changes**
- Fixing bug related to cli flag `-q` (`--queries-path`)
- Adding new cli flag `-b` (`--library`):  path to directory with libraries (default "./assets/libraries"). it assumes users pass a directory with libraries named as <platform>.rego

TODO: Correct E2E tests when I am back

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: pkg/engine/inspector.go [hunk 1]
```diff
@@ -137,14 +137,25 @@ func NewInspector(
...
+			if commonGeneralQuery != "" {
+				opaQuery, err = rego.New(
+					rego.Query(regoQuery),
+					rego.Module("Common", commonGeneralQuery),
+					rego.Module("Generic", platformGeneralQuery),
+					rego.Module(metadata.Query, metadata.Content),
+					rego.Store(store),
+					rego.UnsafeBuiltins(unsafeRegoFunctions),
+				).PrepareForEval(ctx)
+			} else {
```
Comment (rogeriopeixotocx) [inline pkg/engine/inspector.go:150]: should we check if platform-specific queries are not available too with `platformGeneralQuery != ""`

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
@rafaela-soares shouldn't we move our libraries from this path format: `/assets/libraries/platform/library.rego` to this format: `/assets/libraries/platform.rego`?

Also, I don't see the documentation additions that we discussed.

File: test/main_test.go [hunk 2]
```diff
@@ -217,7 +217,7 @@ func sliceContains(s []string, str string) bool {
 }
 
 func readLibrary(platform string) (string, error) {
-	pathToLib := source.GetPathToLibrary(platform, "../")
+	pathToLib := source.GetPathToLibrary(platform, "../", "./assets/libraries")
```
Comment (felipe-avelar) [inline test/main_test.go:220]: Please, use `filepath.FromSlash` here to prevent cross OS issues

File: pkg/engine/inspector_test.go [hunk 3]
```diff
@@ -595,15 +595,15 @@ type mockSource struct {
...
+	sources := source.NewFilesystemSource(m.Source, []string{""}, []string{""}, "./assets/libraries")
 
 	return sources.GetQueries(queryFilter)
 }
 
 func (m *mockSource) GetQueryLibrary(platform string) (string, error) {
 	currentWorkdir, _ := os.Getwd()
 
-	pathToLib := source.GetPathToLibrary(platform, currentWorkdir)
+	pathToLib := source.GetPathToLibrary(platform, currentWorkdir, "./assets/libraries")
```
Comment (felipe-avelar) [inline pkg/engine/inspector_test.go:606]: Please, use `filepath.FromSlash` here to prevent cross OS issues

File: pkg/engine/inspector_test.go [hunk 4]
```diff
@@ -595,15 +595,15 @@ type mockSource struct {
 }
 
 func (m *mockSource) GetQueries(queryFilter *source.QueryInspectorParameters) ([]model.QueryMetadata, error) {
-	sources := source.NewFilesystemSource(m.Source, []string{""}, []string{""})
+	sources := source.NewFilesystemSource(m.Source, []string{""}, []string{""}, "./assets/libraries")
```
Comment (felipe-avelar) [inline pkg/engine/inspector_test.go:598]: Please, use `filepath.FromSlash` here to prevent cross OS issues

File: pkg/engine/inspector_test.go [hunk 5]
```diff
@@ -572,7 +572,7 @@ func TestEngine_GetFailedQueries(t *testing.T) {
 }
 
 func newInspectorInstance(t *testing.T, queryPath string) *Inspector {
-	querySource := source.NewFilesystemSource(queryPath, []string{""}, []string{""})
+	querySource := source.NewFilesystemSource(queryPath, []string{""}, []string{""}, "./assets/libraries")
```
Comment (felipe-avelar) [inline pkg/engine/inspector_test.go:575]: Please, use `filepath.FromSlash` here to prevent cross OS issues

File: internal/console/scan.go [hunk 6]
```diff
@@ -384,6 +387,11 @@ func initPathsFlags(scanCmd *cobra.Command) {
 		"./assets/queries",
 		"path to directory with queries",
 	)
+	scanCmd.Flags().StringVarP(&libraryPath,
+		libraryFlag, libraryShorthand,
+		"./assets/libraries",
```
Comment (felipe-avelar) [inline internal/console/scan.go:392]: Please, use `filepath.FromSlash` here to prevent cross OS issues

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

**Review by rogeriopeixotocx — DISMISSED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]

**Pull request merged**